test/mitmproxy/addons/test_maplocal.py

```diff
@@ -1,35 +1,59 @@
+import re
+from pathlib import Path
+
 import pytest
 
-from mitmproxy.addons.maplocal import MapLocal, file_candidates
+from mitmproxy.addons.maplocal import MapLocal, MapLocalSpec, file_candidates
+from mitmproxy.addons.modifyheaders import parse_spec
 from mitmproxy.test import taddons
 from mitmproxy.test import tflow
 
-from mitmproxy.addons.modifyheaders import parse_modify_spec
-
 
 @pytest.mark.parametrize(
-    "url,spec,expected_candidates", [
-        (
-            "https://example.org/img/topic/subtopic/test.jpg",
-            ":example.com/foo:/tmp",
-            ["/tmp/img/topic/subtopic/test.jpg", "/tmp/img/topic/test.jpg", "/tmp/img/test.jpg", "/tmp/test.jpg"]
-        ),
+    "url,spec,expected_candidates",
+    [
+        # trailing slashes
+        ("https://example.com/foo", ":example.com/foo:/tmp", ["/tmp/index.html"]),
```

Comment:
Shouldn't this allow that `example.com/foo`is a file that I want to map to /tmp/foo?

Reply:
Then you should specify `:example.com/foo:/tmp/foo` or  `:example.com/:/tmp/foo`, no?

Reply:
I think it should be possible to specify `:/foo/bar/baz.jpg:/tmp/baz.jpg` to map `https://example.com/foo/bar/baz.jpg` to `/tmp/baz.jpg`.

Reply:
Why doesn't that work? That's a single file, so it shouldn't go through file_candidates at all.

Reply:
My bad, you're right. I tested the patterns with the `test_file_candidates` method.